src/prefect/utilities/threaded_heartbeat.py

```diff
@@ -0,0 +1,48 @@
+import threading
+from contextlib import contextmanager
+
+import prefect
+from prefect import config
+from prefect.client import Client
+from prefect.utilities.logging import get_logger
+
+
+class HeartbeatThread(threading.Thread):
```

Comment:
I think this might be easier to read as a function inlined into your `threaded_heartbeat` function. Keeping the implementation together seems nice and subclassing creates confusion about which attrs are for your code and which are for `threading.Thread`


```
def threaded_heartbeat(...):

    def send_heartbeat_thread(stop_event, flow_run_id):
         # contents of run here
    
    thread = threading.Thread(send_heartbeat_thread)
    
```

Reply:
In my recent cleanup I played with writing it out with an inlined `Thread` and a `target` with a handful of arguments, but it ended up just looking confusing to my eyes. `Thread` was designed to be subclassed at times, according to the docs, so if it doesn't feel too wrong for you, would you mind if I kept it this way?